src/Gamecube.h

```diff
@@ -88,6 +88,180 @@ typedef union{
     };
 } Gamecube_Report_t;
 
+typedef union{
+    // 8 bytes of datareport that we get from the controller
+    uint8_t raw8[8];
+    uint16_t raw16[0];
+    uint32_t raw32[0];
+
+    struct{
+        uint8_t buttons0;
+        union{
+            uint8_t buttons1;
+            uint8_t dpad : 4;
+        };
+    };
+
+    struct {
+        // first data byte (bitfields are sorted in LSB order)
+        uint8_t a : 1;
+        uint8_t b : 1;
+        uint8_t x : 1;
+        uint8_t y : 1;
+        uint8_t start : 1;
+        uint8_t origin : 1; // Indicates if GetOrigin(0x41) was called (LOW)
+        uint8_t errlatch : 1;
+        uint8_t errstat : 1;
+
+        // second data byte
+        uint8_t dleft : 1;
+        uint8_t dright : 1;
+        uint8_t ddown : 1;
+        uint8_t dup : 1;
+        uint8_t z : 1;
+        uint8_t r : 1;
+        uint8_t l : 1;
+        uint8_t high1 : 1;
+
+        // 3rd-8th data byte
+        uint8_t xAxis;
+        uint8_t yAxis;
+        uint8_t cxAxis;
+        uint8_t cyAxis;
+        uint8_t left;
+        uint8_t right;
+    };
```

Comment:
I think it would make sense to add a comment here, that this is mode 3. Do not give it a name, but say maybe:
```
}; // mode3 (default reading mode)
```